lib/admin/components/UserRow.js

```diff
@@ -154,78 +151,120 @@ export default class UserRow extends Component<Props, State> {
       return null
     }
     const userEmailDataSlug = user.email.split('@')[0]
-
     const userSettings = getSettingsFromProfile(user)
+    const accountType = userSettings && userSettings.account_type
+
+    const accountTypeKeys = Object.keys(accountTypes)
+    const hasDefaultKey = accountTypeKeys.includes('default')
+    const accountTypeUnknown = accountType && !accountTypeKeys.includes(accountType)
+
+    const isDefault = accountType === 'default'
+    const displayName = !accountType || accountTypeUnknown
+      ? hasDefaultKey
+        ? accountTypes.default.name
+        : `${this.messages('unknownAccount')} (${accountType || ''})`
+      : accountTypes[accountType].name
 
     return (
-      <ListGroupItem>
-        <Row className='list-group-item-heading'>
-          <Col xs={4} sm={2} md={1}>
+      <>
+        <tr>
+          <td>
             <Image
-              alt={user.email}
-              responsive
+              alt='Avatar'
               rounded
               src={user.picture}
+              style={{ width: '42px' }}
             />
-          </Col>
-          <Col xs={8} sm={5} md={6}>
-            <h5>
-              {user.email}{' '}
-              {this._getUserPermissionLabel(permissions)}
-              {' '}
-              {this._getOrgLabel(permissions)}
-            </h5>
-            <small />
-          </Col>
-          <Col xs={12} sm={5} md={5}>
+          </td>
+          <td>
+            {user.email}
+          </td>
+          <td>
+            {permissions.isApplicationAdmin()
+              ? <BsLabel bsStyle='danger'>{this.messages('appAdmin')}</BsLabel>
+              // Display non-admin account types if more than one are configured.
+              : accountTypeKeys.length > 1 && (
+                <BsLabel bsStyle={accountTypeUnknown ? 'warning' : isDefault ? 'default' : 'primary'}>{displayName}</BsLabel>
+              )
+            }
+          </td>
+          <td>
+            {!permissions.isApplicationAdmin() && this._getUserPermissionLabel(permissions)}
+            {' '}
+            {this._getOrgLabel(permissions)}
+          </td>
+          {/* When not editing, show the "Last login" cell and
+              show the "actions" column with just the edit button.
+              When editing, hide the "Last login" cell and
+              expand the "actions" column over the space where last login is shown,
+              so that there is enough space to show the delete/save/cancel buttons. */}
+          <td style={isEditing ? { display: 'none' } : null}>
```

Comment:
Why do we use `display: none` instead of just not rendering the element?

Reply:
My intuition was to keep the DOM tree the same instead of adding/removing an element on toggle.

Reply:
Wouldn't removing the element be the more "react-ey" way to do it? There may be performance implications but I think within datatools these are ignorable.

Reply:
Updated in https://github.com/ibi-group/datatools-ui/pull/859/commits/60b036c9179c7974ca48bf4c5c7087deb2fe05fa.